### PR TITLE
Deprecate -c cli short variant flag in docker cli build & create

### LINKED ISSUE #17117: Deprecate '-c' short variant flag for 'docker build' and 'docker create'
As I say on #17101.

Given that `-c` (cpu-shares) flag as been removed from `docker run` in #16271, I think it should also be removed from `docker build` and `docker create` for consistency.

### PR DESCRIPTION
**Pull request by vdemeester**
Fixes #17117 🐻.

Deprecates `-c` (cpu-shares) flag for `docker build` and `docker create` and updates docs, and contrib completion.
- build
- create

/ping @runcom @tiborvass @sdurrheimer @albers @thaJeztah 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

**Comment by tiborvass** [comment]
code lgtm

I don't know about docs still mentioning `-c` even if deprecated.

**Comment by vdemeester** [comment]
@tiborvass yeah me neither >_<. @thaJeztah WDYT ? :pray:

**Comment by calavera** [comment]
LGTM

**Comment by vdemeester** [comment]
Oh waw.. needs a rebase :sweat_smile:

**Comment by thaJeztah** [comment]
gee, will have to look that up, that'll be fun "search: 'c'"

File: man/docker-build.1.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ set as the **URL**, the repository is cloned locally and then sent as the contex
 **--memory-swap**=*MEMORY-SWAP*
   Total memory (memory + swap), '-1' to disable swap.
 
-**-c**, **--cpu-shares**=*0*
+**-c** (deprecated), **--cpu-shares**=*0*
```
Comment (thaJeztah) [inline man/docker-build.1.md]: I think we should just remove them here; we have marked them deprecated on the deprecation page; doing it here again isn't needed.

File: man/docker-create.1.md [hunk 2]
```diff
@@ -83,7 +83,7 @@ The initial status of the container created with **docker create** is 'created'.
 **--blkio-weight**=0
    Block IO weight (relative weight) accepts a weight value between 10 and 1000.
 
-**-c**, **--cpu-shares**=0
+**-c** (deprecated), **--cpu-shares**=0
```
Comment (thaJeztah) [inline man/docker-create.1.md]: same here

File: man/docker-run.1.md [hunk 3]
```diff
@@ -100,14 +100,14 @@ option can be set multiple times.
 **--blkio-weight**=0
    Block IO weight (relative weight) accepts a weight value between 10 and 1000.
 
-**-c**, **--cpu-shares**=0
+**-c** (deprecated), **--cpu-shares**=0
```
Comment (thaJeztah) [inline man/docker-run.1.md]: and here

File: man/docker-build.1.md (hunk 1 above)
Comment (vdemeester) [inline man/docker-build.1.md]: Ok, perfect :heart_eyes_cat:

**Comment by thaJeztah** [comment]
Searching for other docs that still mention the `-c`;
https://github.com/docker/docker/blob/master/docs/reference/commandline/build.md
https://github.com/docker/docker/blob/master/docs/reference/commandline/create.md

https://github.com/docker/docker/blob/c45ad0b02d1cc1f88aaabab18aed27aeb20b9e7a/docs/reference/commandline/cli.md#help

Some leftovers from "run"
https://github.com/docker/docker/blob/2b4128e904cf5ffc288e45dbc049de73a905e4e6/docs/reference/commandline/run.md#L20

**Comment by tiborvass** [comment]
:heart: @thaJeztah

**Comment by thaJeztah** [comment]
It was less problematic than I expected, luckily :D

**Comment by vdemeester** [comment]
Now I want a T-Shirt written "I :heart: @thaJeztah" :wink:

File: docs/reference/commandline/build.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ parent = "smn_cli"
     Build a new image from the source code at PATH
 
       --build-arg=[]                  Set build-time variables
-      -c, --cpu-shares                CPU Shares (relative weight)
+      --cpu-shares                CPU Shares (relative weight)
```
Comment (vdemeester) [inline docs/reference/commandline/build.md]: Ah.. messed up the formating here :sweat:

**Comment by thaJeztah** [comment]
OMG LOL

**Comment by thaJeztah** [comment]
LGTM, thanks @vdemeester

**Comment by runcom** [comment]
:tada: thx @vdemeester

**Comment by sdurrheimer** [comment]
:+1: @vdemeester

**Pull request merged**